### PR TITLE
Remove unnecessary request data preparation

### PR DESCRIPTION
**Pull request by marjev**

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=962&repo=edx%2Fecommerce

**Comment by marjev** [comment]
@vkaracic @iivic Please review this.

**Review by vkaracic — APPROVED** [review]
LGTM

File: ecommerce/extensions/catalogue/utils.py [hunk 1]
```diff
@@ -171,3 +113,61 @@ def create_coupon_product(
...
+
+    for catalog in catalogs:
+        if catalog.name == name and catalog.partner == partner:
+            if set(catalog.stock_records.all()) == set(stock_records):
+                return catalog, False
+
+    catalog = Catalog.objects.create(name=name, partner=partner)
+    for stock_record in stock_records:
+        catalog.stock_records.add(stock_record)
+    return catalog, True
```
Comment (vkaracic) [inline ecommerce/extensions/catalogue/utils.py:173]: Omg, you alphabetized this.

**Review by iivic — APPROVED** [review]

**Pull request merged**